src/ObjCRuntime/Runtime.cs

```diff
@@ -158,6 +161,15 @@ internal unsafe struct InitializationOptions {
 					return (Flags & InitializationFlags.IsSimulator) == InitializationFlags.IsSimulator;
 				}
 			}
+
+#if NET
+			// Future optimization potential: make the linker change this into a constant.
+			internal bool IsCoreCLR {
+				get {
+					return (Flags & InitializationFlags.IsCoreCLR) == InitializationFlags.IsCoreCLR;
```

Comment:
nit: I like to use https://docs.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=net-5.0 easier for new developers and does the same.

Reply:
Not that it really matters (this property isn't going to be called a billion times), but it's a lot slower...

```
$ dotnet run
TestHasFlag: 20686
InLine: 2635
```

Test case: https://gist.github.com/rolfbjarne/d75f107521a0d7a3e452df7ad7ab8904

For curiosity I tried mono as well, and mono is much faster than .NET in this case:

```
$ csc *.cs /nologo && mono *.exe
TestHasFlag: 1808
InLine: 973
```

Reply:
What are they doing to make a flag check slow??????

Reply:
That's really weird. The call should be "intrinsic" in the runtime, for .NET 6 with both MonoVM and CoreCLR and even in interpreter (iirc). You are likely hitting the JIT tiering. The microbenchmarks with BenchmarkDotNet easily fix that measurement mistake (and there's a `dotnet new benchmark` template so there's no excuse to not use it; https://github.com/dotnet/BenchmarkDotNet/blob/10abc4f4daa5142f0257c9d512a38afc2ab28be1/docs/articles/guides/dotnet-new-templates.md).

Reply:
To settle the debate on (value & E.A) == E.A vs value.HasFlag(E.A) I run the benchmark with BDN:
```
|      Method |   Mean |  Error |
|------------ |-------:|-------:|
| TestHasFlag | 0.0 ns | 0.0 ns |
|  TestInLine | 0.0 ns | 0.0 ns |
```
(netcoreapp3.1, running on M1 under Rosetta)

That said, on the first run it measured some difference (but likely a fluke). CoreCLR should generate exactly the same code for both so you will likely observe memory access patterns, tiering and other stuff. Mono also implements special codegen for `HasFlag`. IL is slightly bigger for `HasFlag`, machine code should be identical.

Reply:
And obligatory Sharplab.io comparison: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBoDtt8ZcAHbMGAAgAs8Y6ArvgD6wfmG40A3jU5zOxAMwLSC8gHZZ8mdQCQ8zgG0AYgBtsAc1wBdGvvn8hnAKKcdB3boCCnALydyNDsPACE/TlIg6gMDAF8aLRjOFigASwA3bAwuV0zTAS5/ZwA6LwBuBOikhWVgCAhTTgAVZgwACTwzSwAKAEo/AD5OPILijtwui26Sr16K2j1dJU46hubWgEk6ABlUuhg+wc5uka4AMhdS/t8i0vm4mnjqIA==

Looks like there may be small difference in addressing but that's quite specific to the actual usage and may disappear on inlining. (/cc @EgorBo)

Reply:
@filipnavara that redundant move should gone once jit gets "forward substitution" :) 

@rolfbjarne regarding your benchmark, it seems inliner doesn't want to inline `TestHasFlag` - if you help it by adding `[MethodImpl(MethodImplOptions.AggressiveInlining)]` it should be the same (there will be a small difference - but if you re-order your stopwatches - `TestHasFlag` will be faster)